op.c

```diff
@@ -16477,24 +16481,81 @@ Perl_rcpv_copy(pTHX_ char *pv) {
 
 /* Subroutine signature parsing */
 
+struct yy_parser_signature_param {
+    UV         argix;        /* positional index of the param */
+    PADOFFSET  padix;        /* pad index of the var holding the param */
+    OP        *defcop;
+    OPCODE     defmode;
+    OP        *defexpr;
+};
+
 struct yy_parser_signature {
-    UV          elems;      /* number of signature elements seen so far */
-    UV          optelems;   /* number of optional signature elems seen */
+    /* A note on terminology. We call each variable that appears in a
+     * signature a "parameter", and each scalar value the caller passes in at
+     * call time we call an "argument".
+     * Each scalar parameter variable will correspond to one argument value.
+     * The complication arises when we consider a slurpy array or hash
+     * parameter. Here, while it is exactly one *parameter*, it may account
+     * zero or more arguments.
+     */
+    UV          next_argix; /* the argument index of the next parameter we add */
+    UV          opt_params; /* number of optional scalar parameters */
     char        slurpy;     /* the sigil of the slurpy var (or null) */
-    OP         *elemops;    /* NULL, or an OP_LINESEQ of individual element and fence ops */
```

Comment:
move field  char slurpy; to the bottom of the struct.  don't have hidden alignment holes.